server/src/views/profile.js

```diff
@@ -0,0 +1,249 @@
+/* jshint browser:true */
+
+const React = require("react");
+const errors = require("../../shared/errors");
+
+function delay(duration) {
+  return new Promise(resolve => setTimeout(resolve, duration));
+}
+
+exports.ProfileButton = class ProfileButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onDocumentClick = this.dismiss.bind(this);
+    this.onRefreshProfileBound = this.onRefreshProfile.bind(this);
+    this.state = {
```

Comment:
In clientglue.js I think it should keep the profile information in the model that it uses for render().  Then clientglue is what catches the profile-update event from viewerworker, and then rerenders the entire page.